app/uk/gov/hmrc/tai/viewModels/taxCodeChange/TaxCodeChangeViewModel.scala

```diff
@@ -23,14 +23,29 @@ import uk.gov.hmrc.tai.model.domain.{TaxCodeChange, TaxCodeRecord}
 import uk.gov.hmrc.tai.util.TaiConstants
 import uk.gov.hmrc.tai.viewModels.{DescriptionListViewModel, TaxCodeDescriptor}
 
-case class TaxCodeChangeViewModel(pairs: TaxCodePairs, changeDate: LocalDate, scottishTaxRateBands: Map[String, BigDecimal])
+case class TaxCodeChangeViewModel(pairs: TaxCodePairs,
+                                  changeDate: LocalDate,
+                                  scottishTaxRateBands: Map[String, BigDecimal],
+                                  gaDimensions: Option[Map[String, String]])
 
 object TaxCodeChangeViewModel extends TaxCodeDescriptor {
-  def apply(taxCodeChange: TaxCodeChange, scottishTaxRateBands: Map[String, BigDecimal])(implicit messages: Messages): TaxCodeChangeViewModel = {
+
+  def apply(taxCodeChange: TaxCodeChange, scottishTaxRateBands: Map[String, BigDecimal]): TaxCodeChangeViewModel = {
+    def moreThanTwoSecondaryWithoutPayroll(records: Seq[TaxCodeRecord]): Boolean = records.count(record => !record.primary && record.payrollNumber.isEmpty) >= 2
```

Comment:
Is this really what we want.

Is not the moreThanTwoSecondaryWithoutPayrollWithSameEmployerName?

Reply:
So same name also? True will update the requirements on the tickets

Reply:
may be worth checking that this is ok